scanpy/readwrite.py

```diff
@@ -277,15 +277,53 @@ def _read_v3_10x_h5(filename, *, start=None):
                 (data, dsets['indices'], dsets['indptr']),
                 shape=(N, M),
             )
+            obs_dict = dict(obs_names=dsets['barcodes'].astype(str))
+            var_dict = dict(
+                var_names=dsets['name'].astype(str),
+            )
+
+            if 'gene_id' not in dsets:
+                # Read metadata specific to a feature-barcode matrix
+                var_dict.update(
+                    {
+                        'gene_ids': dsets['id'].astype(str),
+                    }
+                )
+            else:
+                # Read metadata specific to a probe-barcode matrix
+                var_dict.update(
+                    {
+                        'gene_ids': dsets['gene_id'].astype(str),
+                        'probe_ids': dsets['id'].astype(str),
+                    }
+                )
+            var_dict.update({'feature_types': dsets['feature_type'].astype(str)})
+            if 'filtered_barcodes' in f['matrix']:
+                obs_dict.update(
+                    {'filtered_barcodes': dsets['filtered_barcodes'].astype(bool)}
+                )
+            if 'features' in f['matrix']:
+                var_dict.update(
+                    {
+                        x: dsets[x].astype(bool if y.dtype.kind == 'b' else str)
```

Comment:
```suggestion
                        (x, dsets[x].astype(bool if y.dtype.kind == 'b' else str))
```
Use a generator rather than a comprehension.

Reply:
Good idea. Fixed.